luigi/scheduler.py

```diff
@@ -840,6 +842,10 @@ def add_task(self, task_id=None, status=PENDING, runnable=True,
         if task is None or (task.status != RUNNING and not worker.enabled):
             return
 
+        # if the task was completed only recently, do not run the same task again
+        if status == PENDING and task.status == DONE and (time.time() - task.updated) < self._config.rerun_task_blocking_period:
+            status = DONE
```

Comment:
What happens if you just return here instead???

Reply:
As far as I understand, the same thing happens -- the scheduler rejects giving task to the worker. Changed to simple "return".